### PR TITLE
Add E2E tests and example for VRFs

### PR DESCRIPTION
**Pull request by andreaskaris**
**Is this a BUG FIX or a FEATURE ?**:
/kind enhancement

**What this PR does / why we need it**:
Create E2E tests and add an example for VRFs to make sure that the feature works.

**Special notes for your reviewer**:
Reported-at: https://github.com/nmstate/kubernetes-nmstate/issues/847
Fixes: https://github.com/nmstate/kubernetes-nmstate/issues/847

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/handler/utils.go [hunk 1]
```diff
@@ -196,8 +196,7 @@ func updateDesiredStateWithCaptureAtNodeAndWait(node string, desiredState nmstat
 	policy.WaitForAvailableTestPolicy()
 }
 
-// TODO: After we implement policy delete (it will cleanUp desiredState) we have
-//       to remove this
+// TODO: After we implement policy delete (it will cleanUp desiredState) we have to remove this.
```
Comment (andreaskaris) [inline test/e2e/handler/utils.go:199]: Note: gofmt does not like the earlier formatting with indentation and messed up the format of this comment, so moving it to one line to make gofmt happy

File: test/e2e/handler/states.go [hunk 2]
```diff
@@ -215,6 +215,32 @@ func vlanAbsent(iface, vlanID string) nmstate.State {
...
+// vrfAbsent takes a VRF identifier and a list of interfaces to remove from that VRF.
+func vrfAbsent(vrfID string, ifaceList ...string) nmstate.State {
+	ifaces := strings.Join(ifaceList, ",")
+	return nmstate.NewState(fmt.Sprintf(`interfaces:
+    - name: vrf%s
+      type: vrf
+      state: absent
+      vrf:
+        port: [%s]
+        route-table-id: %[1]s
```
Comment (qinqon) [inline test/e2e/handler/states.go:240]: To remove it we don't need this part

File: test/e2e/handler/static_addr_and_route_test.go [hunk 3]
```diff
@@ -26,7 +26,11 @@ import (
 	nmstate "github.com/nmstate/kubernetes-nmstate/api/shared"
 )
 
-func ipV4AddrAndRoute(firstSecondaryNic, ipAddress, destIPAddress, prefixLen, nextHopIPAddress string) nmstate.State {
+func ipV4AddrAndRoute(firstSecondaryNic, ipAddress, destIPAddress, prefixLen, nextHopIPAddress,
```
Comment (qinqon) [inline test/e2e/handler/static_addr_and_route_test.go:29]: Let's keep `ipV4AddrAndRoute` and add a new `ipV4AddrAndRouteẀithTableID`, the first one will call the second one with tableID "". Test without TableID are more common.

File: test/e2e/handler/utils.go [hunk 4]
```diff
@@ -612,9 +618,14 @@ func routeDest(node, destIP string) string {
 	return gjson.ParseBytes(currentStateJSON(node)).Get(path).String()
 }
 
-func routeNextHopInterface(node, destIP string) AsyncAssertion {
+// routeNextHopInterface checks if a route with destIP exists in table tableID. If tableID is the empty string, use
+// the default table-id (254).
+func routeNextHopInterface(node, destIP, tableID string) AsyncAssertion {
```
Comment (qinqon) [inline test/e2e/handler/utils.go:623]: Same here let's keep `routeNextHopInterface` and create new `routeNextHopInterfaceWithTableID`

**Review by qinqon — CHANGES_REQUESTED** [review]
Nits about wrapping functions so we don't affect the other tests.

File: test/e2e/handler/states.go (hunk 2 above)
Comment (andreaskaris) [reply test/e2e/handler/states.go]: Done thanks

File: test/e2e/handler/static_addr_and_route_test.go (hunk 3 above)
Comment (andreaskaris) [reply test/e2e/handler/static_addr_and_route_test.go]: Yep looks better, done

File: test/e2e/handler/utils.go (hunk 4 above)
Comment (andreaskaris) [reply test/e2e/handler/utils.go]: Yep looks better, done

**Review by qinqon — CHANGES_REQUESTED** [review]
nit doc: Can you add the example to https://github.com/nmstate/kubernetes-nmstate/blob/main/docs/examples.md ? the examples index is not dynamic

**Comment by andreaskaris** [comment]
> nit doc: Can you add the example to https://github.com/nmstate/kubernetes-nmstate/blob/main/docs/examples.md ? the examples index is not dynamic

done :-)

**Comment by qinqon** [comment]
The example is present at generated docs 
https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/1189/pull-kubernetes-nmstate-docs/1663472420981313536/artifacts/gh-pages/examples.html#vrf

/lgtm
/approve

**Pull request merged**